### PR TITLE
Add magnifier to text selection & better platform-specific behavior of text selection

### PR DESCRIPTION
**Pull request by amantoux**
- Adds magnifier upon selecting text on mobile
- Android/iOS text selection experience is closer to native
- Spell checking operational but missing squiggles

### CODE REVIEW

**Comment by amantoux** [comment]
Closes #198

File: packages/fleather/lib/src/widgets/text_selection.dart [hunk 1]
```diff
@@ -1,6 +1,4 @@
-// Copyright 2014 The Flutter Authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
+// ignore_for_file: deprecated_member_use_from_same_package
```
Comment (Amir-P) [inline packages/fleather/lib/src/widgets/text_selection.dart:1]: Dragging collapsed selection handle opens magnifier both in native widgets and Flutter's `TextField` but not in current implementation of this PR.

**Review by Amir-P — CHANGES_REQUESTED** [review]
Thanks for doing this @amantoux! PTAL at my comments.

File: packages/fleather/lib/src/widgets/editor.dart [hunk 2]
```diff
@@ -36,12 +37,30 @@ typedef FleatherContextMenuBuilder = Widget Function(
...
-        BuildContext context, EditorState editableTextState) =>
+        BuildContext context, EditorState editorState) =>
     AdaptiveTextSelectionToolbar.buttonItems(
-      buttonItems: editableTextState.contextMenuButtonItems,
-      anchors: editableTextState.contextMenuAnchors,
+      buttonItems: editorState.contextMenuButtonItems,
+      anchors: editorState.contextMenuAnchors,
     );
 
+Widget defaultSpellCheckMenuBuilder(
```
[the remark this answers is not in the document]
Comment (amantoux) [reply packages/fleather/lib/src/widgets/editor.dart:46]: Let me check, I might have misread as I thought `TextField` did not support spell check on other platforms

File: packages/fleather/lib/src/widgets/text_selection.dart (hunk 1 above)
Comment (amantoux) [reply packages/fleather/lib/src/widgets/text_selection.dart:1]: nice catch, let me check

File: packages/fleather/lib/src/widgets/editor.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (amantoux) [reply packages/fleather/lib/src/widgets/editor.dart:46]: Spell check service throws

`Spell check was enabled with spellCheckConfiguration, but the current platform does not have a supported spell check service, and none was provided. Consider disabling spell check for this platform or passing a SpellCheckConfiguration with a specified spell check service.` on desktop

Other platforms required ad hoc widgets

**Review by Amir-P — APPROVED** [review]
LGTM!

**Pull request merged**